src/js/workshop-comments.js

```diff
@@ -124,15 +124,19 @@ function disableFields(disabled) {
 	// When a UIO theme is selected, the background color will remain the same as what is set by the theme.
 	const backgroundColor = disabled ? "#f3f3f3" : "#fff";
 
-	// Disable/enable and set background color of post a comment button
+	// Disable/enable and set background color of post a comment button.
 	const postCommentButton = document.getElementById("post-comment");
 	postCommentButton.setAttribute("style", "background-color: " + backgroundColor);
```

Comment:
The logic of line 129- 132 and line 137-140 are pretty much the same, please create a function to share.

Reply:
Thanks for creating `reduceOpacity()` function. But what I meant was to share the entire logic of disabling rather than sharing one single action. Looking further into the code, it's even unnecessary to use separate blocks to disable a button and input fields. Line 136-150 can be simplified to:

```
	const commentFormElms = document.querySelectorAll("#name, #comment, #post-comment");
	commentFormElms.forEach( element => {
		element.setAttribute("style", "background-color: " + backgroundColor);
		const elmType = element.nodeName.toLowerCase();
		element[elmType === "button" ? "disabled" : "readOnly"] = disabled;
		// When UIO themes are turned on, reduce the opacity of form elements as a disabled indicator instead of changing the background color.
		// The css class "disable-element" is only defined with UIO themes.
		element.classList[disabled ? "add" : "remove"]("disable-element");
	});
```